sdk/src/timing.rs

```diff
@@ -66,11 +66,12 @@ pub fn slot_duration_from_slots_per_year(slots_per_year: f64) -> Duration {
 }
 
 #[derive(Debug, Default)]
-pub struct AtomicInterval {
-    last_update: AtomicU64,
+pub struct Interval<T> {
```

Comment:
why generic? timestamp will always return u64, right?

Reply:
> Did you measure overhead?

```
running 2 tests
test bench_atomic_interval     ... bench:       4,796 ns/iter (+/- 573)
test bench_non_atomic_interval ... bench:       4,801 ns/iter (+/- 440)
```
non atomic is actually slower.

Reply:
> why generic? timestamp will always return u64, right?

Yes. T is either u64 or AtomicU64 for single thread/multi-thread.